src/holdem/parse.rs

```diff
@@ -522,6 +523,42 @@ impl RangeParser {
 
         Ok(filtered)
     }
+
+    /// Parse a string and return all the starting hands
+    ///
+    /// # Examples
+    ///
+    /// Same as `parse_one` but this will parse a comma separated list of hands.
+    ///
+    /// ```
+    /// use rs_poker::holdem::RangeParser;
+    ///
+    /// let hand = RangeParser::parse_many("KK+,A2s+").unwrap();
+    /// assert_eq!(60, hand.len());
+    ///
+    /// // Filters out duplicates.
+    /// assert_eq!(RangeParser::parse_many("AK-87s,A2s+").unwrap().len(), 72)
+    /// ```
+    pub fn parse_many(r_str: &str) -> Result<Vec<Hand>, Vec<RSPokerError>> {
+        let mut errors: Vec<RSPokerError> = vec![];
+        let mut set = HashSet::new();
+
+        let hands: Vec<_> = r_str
+            .split(',')
+            .map(|s| RangeParser::parse_one(s.trim()))
+            .filter_map(|r| r.map_err(|e| errors.push(e)).ok())
+            .flatten()
+            .collect();
+
+        match errors.is_empty() {
+            true => {
+                set.extend(hands);
+
+                Ok(set.into_iter().collect())
+            }
+            false => Err(errors),
+        }
+    }
```

Comment:
We should only return RSPokerError. What do you think about this?

```suggestion
    pub fn parse_many(r_str: &str) -> Result<Vec<Hand>, RSPokerError> {
        let all_hands: Vec<_> = r_str
            // Split into different ranges
            .split(',')
            // Try to parse the ranges.
            .map(|s| RangeParser::parse_one(s.trim()))
            // Use FromIterator to get a result and unwrap it.
            .collect::<Result<Vec<_>, _>>()?;

        // Filter the unique hands.
        let unique_hands: HashSet<Hand> = all_hands.into_iter().flatten().collect();

        // Transform hands into a vec for storage
        Ok(unique_hands.into_iter().collect())
    }
```